allensdk/brain_observatory/behavior/behavior_project_cache/project_apis/data_io/behavior_ecephys_project_cloud_api.py

```diff
@@ -0,0 +1,336 @@
+import pandas as pd
+from typing import Union, Optional
+from pathlib import Path
+import logging
+import semver
+
+from allensdk.brain_observatory.behavior.behavior_session import (
+    BehaviorSession)
+from allensdk.api.cloud_cache.cloud_cache import (
+    S3CloudCache, LocalCache, StaticLocalCache)
+
+from allensdk.brain_observatory.ecephys.behavior_ecephys_session \
+    import BehaviorEcephysSession
+
+# [min inclusive, max exclusive)
+MANIFEST_COMPATIBILITY = ["0.1.0", "1.0.0"]
+
+
+class BehaviorCloudCacheVersionException(Exception):
+    pass
+
+
+def version_check(manifest_version: str,
+                  data_pipeline_version: str,
+                  cmin: str = MANIFEST_COMPATIBILITY[0],
+                  cmax: str = MANIFEST_COMPATIBILITY[1]):
+    mver_parsed = semver.VersionInfo.parse(manifest_version)
+    cmin_parsed = semver.VersionInfo.parse(cmin)
+    cmax_parsed = semver.VersionInfo.parse(cmax)
+    if (mver_parsed < cmin_parsed) | (mver_parsed >= cmax_parsed):
+        estr = (f"the manifest has manifest_version {manifest_version} but "
+                "this version of AllenSDK is compatible only with manifest "
+                f"versions {cmin} <= X < {cmax}. \n"
+                "Consider using a version of AllenSDK closer to the version "
+                f"used to release the data: {data_pipeline_version}")
+        raise BehaviorCloudCacheVersionException(estr)
+
+
+class BehaviorEcephysProjectCloudApi():
```

Comment:
I think this should be called `VisualBehaviorNeuropixelsProjectCloudApi` for the same reasons I think the project cache class should be called `VisualBehaviorNeuropixelsProjectCache`.

Reply:
renamed